### PR TITLE
17250 - Vite/Vitest Migration

### PR DESCRIPTION
**Pull request by seeker25**
https://github.com/bcgov/entity/issues/17250

### CODE REVIEW

**Comment by seeker25** [comment]
Just waiting to build a new version of FAS-UI, it should pass the last unit test.

**Comment by JazzarKarim** [comment]
Travis, one small nice change that you can do; change "Test with Jest" in ci.yml to "Test with Vitest".

File: auth-web/src/components/auth/account-settings/account-info/AccountInfo.vue [hunk 1]
```diff
@@ -107,19 +107,21 @@
           @update:viewOnlyMode="viewOnlyMode"
         />
 
-        <template v-if="baseAddress" v-can:VIEW_ADDRESS.hide>
```
Comment (JazzarKarim) [inline auth-web/src/components/auth/account-settings/account-info/AccountInfo.vue:110]: I'm guessing Vite didn't like this format? 😆

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2442-rrms77d7.web.app

File: auth-web/src/components/auth/staff/PPRLauncher.vue [hunk 2]
```diff
@@ -39,8 +38,7 @@ export default class PPRLauncher extends Vue {
   }
 
   private getImgUrl (img) {
-    const images = require.context('@/assets/img/')
```
Comment (JazzarKarim) [inline auth-web/src/components/auth/staff/PPRLauncher.vue:42]: Travis, there's a package that fixes this. I probably should've mentioned that to you, my bad. 
https://github.com/bcgov/business-create-ui/blob/3c85ddb387a7dc7b3f94e14194886778ad8f0b15/vite.config.js#L34

File: auth-web/tests/unit/setup.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import Vuex from 'vuex'
+import can from '@/directives/can'
+import { install } from 'vue-demi'
+
+Vue.use(VueTheMask)
+Vue.use(Vuex)
+Vue.use(VueRouter)
+Vue.directive('can', can)
+
+// Need this so composition api works with vue-demi
```
Comment (JazzarKarim) [inline auth-web/tests/unit/setup.ts:14]: Cool!

File: auth-web/vite.config.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  },
+  server: {
+    port: 8080,
+    host: true
+  },
+  test: {
+    globals: true,
+    environment: 'jsdom',
+    setupFiles: './tests/unit/setup.ts',
+    threading: true,
```
Comment (JazzarKarim) [inline auth-web/vite.config.js:68]: Travis, isn't it `threads`? Or is it just the same thing

File: auth-web/src/components/auth/account-settings/account-info/AccountInfo.vue (hunk 1 above)
Comment (seeker25) [reply auth-web/src/components/auth/account-settings/account-info/AccountInfo.vue:110]: useless directive on template

File: auth-web/src/components/auth/staff/PPRLauncher.vue (hunk 2 above)
Comment (seeker25) [reply auth-web/src/components/auth/staff/PPRLauncher.vue:42]: I tried it, didn't work for me

Comment (JazzarKarim) [reply auth-web/src/components/auth/staff/PPRLauncher.vue:42]: Oh OK, I see.

File: auth-web/src/components/auth/account-settings/account-info/AccountInfo.vue (hunk 1 above)
Comment (JazzarKarim) [reply auth-web/src/components/auth/account-settings/account-info/AccountInfo.vue:110]: 👍

File: auth-web/vite.config.js (hunk 4 above)
Comment (seeker25) [reply auth-web/vite.config.js:68]: i'll change it to threads, good catch

**Review by JazzarKarim — APPROVED** [review]
Looks incredible Travis! Awesome, awesome stuff 🙌 Pretty sure you're glad you finally got this out of the way 😆

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2442-rrms77d7.web.app

**Pull request merged**